pandas/core/generic.py

```diff
@@ -9500,6 +9500,11 @@ def _where(
             self._check_inplace_setting(other)
             new_data = self._mgr.putmask(mask=cond, new=other, align=align)
             result = self._constructor(new_data)
+            for i in range(len(result.dtypes)):
```

Comment:
This is the wrong place for this. We can not special case this here

Reply:
Would you mind elaborating on what you mean? I believe it is not a special case since it just affects the type in the case that the inplace is True. Do you mean I should put it within the putmask? @phofl

Reply:
Probably, but this looks already correct on main, so no need to fix I think.

```
1.5.0.dev0+1180.g8c3a2f2ba7
<StringArray>
['1', <NA>, '3']
Length: 3, dtype: string
<class 'pandas._libs.missing.NAType'>
      A
0     1
1  <NA>
2     3
<StringArray>
['1', <NA>, '3']
Length: 3, dtype: string
<class 'pandas._libs.missing.NAType'>
```

Could you simply add a test?

cc @simonjayhawkins

Reply:
oh! sure.

Reply:
> Probably, but this looks already correct on main, so no need to fix I think.

see https://github.com/pandas-dev/pandas/issues/46512#issuecomment-1190151258

I'll confirm the commit where the fix occurred and if we agree that this is the correct behavior, then indeed we just need a test.

Reply:
I think this is essentially the same
As #47628

Reply:
yes from https://github.com/pandas-dev/pandas/issues/47628#issuecomment-1186513802

> But the strange thing is that doing this on the Series level doesn't end up calling StringArray.__setitem__, it seems to go through Series._where and eventually BlockManager.putmask, and ExtensionArray._putmask, and that last one is not correctly implemented for StringArray.

but we should probably have tests for `DataFrame.where` also incase the implementation of `__setitem__` changes to no longer go through `Series._where`

Reply:
Just out of curiosity, how would it be implemented if we should not have special cased it? Is there a link to the change that fixed it on the main by any chance? @phofl